src/olympia/reviewers/serializers.py

```diff
@@ -157,6 +156,13 @@ def get_entry_mime_type(self, entry, blob):
         mime_type = mime.split('/')[0]
         known_types = ('image', 'text')
 
+        if mime_type == 'text':
+            # Allow text mimetypes to be more specific for readable
+            # files. `python-magic`/`libmagic` usually just returns
+            # plain/text but we should use actual types like text/css or
+            # text/javascript.
+            mime, _ = mimetypes.guess_type(entry.name)
+
         return mime, 'binary' if mime_type not in known_types else mime_type
```

Comment:
FWIW I found this one-liner confusing to read - the inline `if` applies to the second part but that wasn't immediately obvious :)

Reply:
I'll keep that in mind when touching it the next time.